### ISSUE TITLE
no action detected in task in included playbook

### ISSUE THREAD

**Issue by COLABORATI** [ansible/ansible#15309]
I have seen that there are multiple issues and all are closed, but I have the same error when including a simple playbook and can not see what is wrong: 

```
no action detected in task
```

the playbook that includes: 

```

---
- name: include play test
  hosts: all

  tasks:
    - include: base.yml

```

the included playbook:

```

---
- hosts: all
  sudo: yes

  tasks:
    - name: apt-get dist-upgrade
      apt: upgrade=dist update_cache=yes

    - name: install basic tools 
      apt: name=apt-transport-https state=latest

```

the included playbook works fine when not included but directly executed. 
ansible is installed from the trusty ppa:

```
root@rails:~# ansible --version
ansible 2.0.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

**Comment by dagwieers** [comment]
You cannot include a complete playbook inside a play. You can include a "taskbook" (a list of tasks) within a play, but not one or more "plays".

However you can include a playbook inside another playbook, but not at the task-level, you have to do it at the play-level.

``` yaml
- hosts: localhost
  gather_facts: no
  tasks:
  - include: files/taskbook.yml

- include: playbook.yml
```

A "taskbook" consists of a list of tasks and would look like this:

``` yaml
- debug: msg='Example 1 from within a taskbook task'
- debug: msg='Example 2 from within a taskbook task'
```

While a playbook consists of a list of plays and would look like this:

``` yaml
- hosts: localhost
  gather_facts: no
  tasks:
  - debug: msg='Example 1 from within a playbook play'

- hosts: localhost
  gather_facts: no
  tasks:
  - debug: msg='Example 2 from within a playbook play'
```

**Comment by bcoca** [comment]
# Possible Misunderstanding

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We believe the ticket you have filed is being somewhat misunderstood, as one thing works a little differently than stated.

In the future, this might be a topic more well suited for the user list, which you can also post here if you'd like some more help with the above.
- https://groups.google.com/forum/#!forum/ansible-project

Thank you once again for this and your interest in Ansible!

**Comment by COLABORATI** [comment]
Thank you very much for taking your time to explain this nifty little detail - this is very appreciated! 

I was not able to conclude from the documentaion that there is anything that can not be included - there is no hint about the fact that there are limitations regarding the include directive. On the opposite I found several sentences like "Playbooks can also include plays from other playbook files." that made me not understand that there are things that can not be included or that a successful inclusion of other files is dependent on the exact positioning of this directive.

Please add this important information to the docs, I would like to suggest to put the example from dagwieers into http://docs.ansible.com/ansible/playbooks_roles.html

Also it would make these things more clear if there was an exact explanation of what can be included and what can not be included and from where exactly what can and what can not be included here: 
http://docs.ansible.com/ansible/glossary.html#includes

Again I would like to suggest to copy from dagwieers explanation in this issue, and of course add other cases that are not yet desribed.

Also it would be very helpful for a better understanding of the subtle nuances of the created terminology to put an exact definition of the terms with code examples (like in this issue) into the glossary. Also, for a better understanding, it would be good to have contrasting examples e.g. of what _is_ a playbook, task, play, role etc. and what _is not_ a playbook, task, play, role etc. in yaml syntax. 

Thanks again for your attention!

**Comment by COLABORATI** [comment]
Ah, and btw the error handling of wrong includes should be reworked - the error message should be something like "wrong usage of include directive at..." to actually tell what is wrong.
Thanks again!

**Comment by dagwieers** [comment]
@COLABORATI Thanks for the feedback ! Much appreciated.

It helps to get a different point of view in order to improve the documentation. Could you create a pull request with the needed changes to the documentation ? In such a way that you would have found it in the documentation and in the exact wording that would have made things more clear for others like you ?

**Comment by COLABORATI** [comment]
Well, I started collecting examples and comparing several pieces of the documentation and in fact my first attempt to my answer was half way to some updates, but the rabbit hole got deeper and deeper and I had to switch back to the real task of the day. Good documentation should be created by a dedicated team, not by random pull requests. Now with RedHat as mothership I hope it will be no problem to build a team that systematically improves the docs.
